src/components/Api.js

```diff
@@ -0,0 +1,89 @@
+export default class Api {
+    constructor(options){
+        this._baseUrl = options.baseUrl;
+        this._headers = options.headers;
+    }
+
+    getUserInfo() {
+        return fetch(`${this._baseUrl}/users/me`, {
+          headers: this._headers
+        })
+        .then(this._checkResponse);
+    }
+    
+    _checkResponse(res) {
+        if(!res.ok){
+            Promise.reject(`Error ${res.status}`)
+        }
+        return res.json();
+    }
+
+    async getInitialCards() {
+        try {
+            const res = await fetch(`${this._baseUrl}/cards`, {
+                headers: this._headers,
+            });
+            if (!res.ok) {
+                throw new Error(`Error fetching cards: ${res.status}`);
+            }
+            return await res.json();
```

Comment:
Please, use `.then(this._checkResponse);` everywhere not to duplicate the same code